.github/workflows/ci.yaml

```diff
@@ -187,3 +187,92 @@ jobs:
         uses: ./.github/workflows/composite/collectlogs
         with:
           artifact-name: kind-logs-${{ matrix.ip-family }}-${{ matrix.deployment }}
+
+  metallb_e2e:
+    runs-on: ubuntu-22.04
+    needs:
+      - unit-tests
+      - build-test-images
+      - commitlint
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install linux-modules-extra-$(uname -r) python3-pip arping ndisc6
+
+      - name: Download frr-k8s images
+        uses: actions/download-artifact@v3
+        with:
+          path: image
+
+      - name: Load image
+        working-directory: image
+        run: |
+          docker load -i image-tar-frrk8s/frrk8s.tar
+
+      - name: Deploy frr-k8s
+        run: |
+          helm_args="\
+          --set prometheus.serviceMonitor.metricRelabelings[0].sourceLabels=\{__name__\} \
+          --set prometheus.serviceMonitor.metricRelabelings[0].regex=\"frrk8s_bgp_(.*)\" \
+          --set prometheus.serviceMonitor.metricRelabelings[0].targetLabel=\"__name__\" \
+          --set prometheus.serviceMonitor.metricRelabelings[0].replacement=\"metallb_bgp_\\\$\$1\" \
+          --set prometheus.serviceMonitor.metricRelabelings[1].sourceLabels=\{__name__\} \
+          --set prometheus.serviceMonitor.metricRelabelings[1].regex=\"frrk8s_bfd_(.*)\" \
+          --set prometheus.serviceMonitor.metricRelabelings[1].targetLabel=\"__name__\" \
+          --set prometheus.serviceMonitor.metricRelabelings[1].replacement=\"metallb_bfd_\\\$\$1\" \
+          "
+          KIND_CLUSTER_NAME="kind" NAMESPACE=metallb-system LOGLEVEL=debug IP_FAMILY="ipv4" HELM_ARGS=${helm_args} make "deploy-helm"
+          mkdir -p /tmp/kind_logs/
+
+      - name: Checkout MetalLB
+        uses: actions/checkout@v4
+        with:
+          repository: metallb/metallb
+          path: metallb
+          ref: "main"
+
+      - name: Deploy MetalLB
```

Comment:
Why not just deploying metallb in frr mode overriding the frrk8s image? What am I missing?

Reply:
I think it would be messy when prs change the helm chart itself (calling the new changes from metallb's chart)

Reply:
I _think_ we could just copy the whole chart folder under metallb's one (which is what we end up doing when we declare frrk8s as a metallb dependency anyway). That'd remove all the logic we are adding here.
Does it make sense? I think it's worth an attempt

Reply:
sure I can try

Reply:
it looks like I made it work, but not sure I like the outcome (the one later where we will use `inv dev-env`) 😅 seems a bit hacky (unless that's not what you meant), and will also require updating dev-env to accept extra helm args + loading more images into kind. wdyt should we continue with this or revert to what I originally did?

Reply:
yep let's revert and see if this won't bite us in the future :P

Reply:
done